@here/harp-mapview/lib/text/MapViewState.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { GeometryKindSet } from "@here/harp-datasource-protocol";
+import { MapView } from "../MapView";
+import { ViewState } from "./ViewState";
+
+/**
+ * View state obtained from a MapView instance.
+ */
+export class MapViewState implements ViewState {
```

Comment:
Wouldn't it be better to have a copy of the values instead of fetching them? I guess it's fine for JavaScript b/c we don't have multi threading and mapview cannot change while we are rendering text.

Reply:
So far I'm creating this object once and passing it to the text renderer in the constructor. So it's meant to be a dynamic object, it gives the state of the view at the moment of the call.

The only reason I see to copy anything there would be as perf improvement if some of the getters are called too often or take too long. So far I haven't see any issue there when profiling.